src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/BindingMetadataProviderContext.cs

```diff
@@ -36,6 +39,16 @@ public class BindingMetadataProviderContext
         public ModelMetadataIdentity Key { get; }
 
         /// <summary>
+        /// Gets the property attributes.
+        /// </summary>
+        public IReadOnlyList<object> PropertyAttributes { get; }
```

Comment:
In theory could avoid "extra storage" and stop copying the ref in the ctor and implement this as something like:

```c#
public IReadOnlyList<object> PropertyAttributes { get; } => attributes.PropertyAttributes;
```